### PR TITLE
feat(gis): Add Extensions, Gestures, & Launcher pages

### LINKED ISSUE #25: Adding extensions page to desktop-widget / GIS

### PR DESCRIPTION
**Pull request by mmstick**
Closes #25

### CODE REVIEW

**Review by jackpot51 — APPROVED** [review]

File: src/gis/gestures.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+use gio::prelude::*;
+use gtk::prelude::*;
+
+pub fn page(header: &gtk::Widget) -> gtk::Widget {
+    let description = gtk::LabelBuilder::new()
+        .wrap(true)
+        .justify(gtk::Justification::Center)
+        .label(concat!(
+            "Use four finger swip left to pen Workspaces and windows overview, four fingers swipe ",
+            "right to pen Applications, and four fingers swipe up or down to switch between ",
```
Comment (digitalcircuit) [inline src/gis/gestures.rs]: Minor typos:
```suggestion
            "Use four finger swipe left to open Workspaces and windows overview, four fingers swipe ",
            "right to open Applications, and four fingers swipe up or down to switch between ",
```

Note: I'm not sure if the additional letters cause the lines to exceed the preferred length.

**Review by digitalcircuit — COMMENTED** [review]
~~Minor typos found - hope this helps!~~  I appreciate the effort to communicate the Extensions upgrade given the unfortunate awkwardness of the GSettings disable key not being feasible within the upgrade environment.

**Edit:**  Fixed!

File: src/gis/gestures.rs (hunk 1 above)
Comment (jackpot51) [reply src/gis/gestures.rs]: Thanks, looks like @mmstick applied this suggestion

File: src/gis/extensions.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    if !extensions_backup.exists() || extensions_notified.exists() {
+        return None;
+    }
+
+    let _ = File::create(extensions_notified);
+
+    let label = fomat!(
+        "Manually installed GNOME Shell extensions are disabled to ensure upgrades are reliable. "
+        "The extensions are not usually tested as part of Pop!_OS and can cause issues. "
+        "You can manually re-enable them, one at a time, to ensure compatibility of each "
```
Comment (jacobgkau) [inline src/gis/extensions.rs]: The commas around `one at a time` are not present in the mockup.

**Comment by jackpot51** [comment]
I see Super key configuration can be changed at any time from the Settings application on the Gestures page

**Comment by jackpot51** [comment]
@mmstick we need to disable the video for now, we will resolve it with correct backgrounds post-release.

**Review by jackpot51 — APPROVED** [review]

**Comment by jacobgkau** [comment]
This has failed to build on the build server and is also failing locally:
```
cargo build --release --frozen
error: no matching package named `fomat-macros` found
```
I see the requested version of that package is [available](https://crates.io/crates/fomat-macros/0.3.1), so not sure what the problem is.

This needs to build so https://github.com/pop-os/gnome-initial-setup/pull/70 can build.

**Comment by jackpot51** [comment]
@jacobgkau strange, I did a clean build just fine

**Comment by jacobgkau** [comment]
After removing and re-cloning, I'm able to build successfully. Doesn't explain why the build server failed, but I'll proceed with testing locally.

**Comment by jackpot51** [comment]
@jacobgkau I scheduled a rebuild.

**Comment by jacobgkau** [comment]
All of the copy and graphics match the mockups with the exception of this:
Screenshot from 2021-06-28 11-50-22
The paths are in quotation marks here, where they are not in quotation marks in the mockup. Looking at the code, it doesn't look like the quotes are being explicitly added. Is there an easy fix for this? If not, it's a minor thing, but it does look a little cluttered with the quotes, and I don't see them adding any value (I still can't select the paths by double-clicking, so they just make it more difficult to select the full path.)

**Comment by mmstick** [comment]
Maybe if I make a separate label for the paths, that'll make it easier to select.

**Comment by mmstick** [comment]
The paths are now being displayed with the `Display` trait rather than `Debug` trait.

**Review by jacobgkau — APPROVED** [review]
Looks just like the mockup now, and the paths can be selected with a triple-click:

Screenshot from 2021-06-28 12-54-41

The paths are not hyperlinked as Jeremy suggested, but if someone doesn't know how to navigate to the directories manually, then they really ought to just install the most up-to-date versions from extensions.gnome.org anyway.

All of the COSMIC settings exposed in GIS work properly. Not seeing any more issues.

**Review by jackpot51 — APPROVED** [review]

**Pull request merged**